src/routes/AppContent/index.tsx

```diff
@@ -51,36 +58,43 @@ const AppContent = () => {
     // Need this to subscribe to advertiser info after user has created an advertiser.
     // setHasCreatedAdvertiser is triggered inside of NicknameModal.
     useEffect(() => {
-        if (hasCreatedAdvertiser) subscribeAdvertiserInfo({});
-    }, [hasCreatedAdvertiser, subscribeAdvertiserInfo]);
+        if (hasCreatedAdvertiser) {
+            // @ts-expect-error passthrough value is not defined in the type
+            subscribeAdvertiserInfo({ passthrough: { id: 'nickname' } });
+            setHasCreatedAdvertiser(false);
+        }
+    }, [hasCreatedAdvertiser, setHasCreatedAdvertiser, subscribeAdvertiserInfo]);
 
     useEffect(() => {
         setActiveTab(getActiveTab(location.pathname));
     }, [location]);
 
-    return (
-        <AdvertiserInfoStateProvider
-            value={{
+    useEffect(() => {
+        if (isSubscribed)
+            setUserInfoState({
                 error,
+                isActive: isSubscribed,
                 isIdle,
                 isLoading,
-                isSubscribed,
-                setHasCreatedAdvertiser,
-            }}
-        >
-            <div className='app-content'>
-                <Text
-                    align='center'
-                    as='div'
-                    className='app-content__title p-2'
-                    size={isDesktop ? 'xl' : 'lg'}
-                    weight='bold'
-                >
-                    Deriv P2P
-                </Text>
-                {isLoadingActiveAccount && !isEndpointRoute ? (
-                    <Loader />
-                ) : (
+            });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isSubscribed]);
```

Comment:
here we just need to listen to isSubscribed for p2p_settings, and then we can set the UserInfoState values instead of passing it down to a provider